cyral/internal/tokensettings/resource.go

```diff
@@ -0,0 +1,65 @@
+package tokensettings
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/cyralinc/terraform-provider-cyral/cyral/client"
+	"github.com/cyralinc/terraform-provider-cyral/cyral/core"
+)
+
+func resourceSchema() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages the access token settings. See also the data source " +
+			"[`cyral_access_token_settings`](../data-source/access_token_settings.md).",
+		CreateContext: core.CreateResource(updateConfig(), readConfig()),
+		ReadContext:   core.ReadResource(readConfig()),
+		UpdateContext: core.UpdateResource(updateConfig(), readConfig()),
+		DeleteContext: resourceAccessTokenSettingsDelete,
+		Schema:        getAccessTokenSettingsSchema(false),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func readConfig() core.ResourceOperationConfig {
+	return core.ResourceOperationConfig{
+		Name:       "AccessTokenSettingsRead",
+		HttpMethod: http.MethodGet,
+		CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+			return fmt.Sprintf("https://%s/v1/accessTokens/settings", c.ControlPlane)
+		},
+		NewResponseData: func(d *schema.ResourceData) core.ResponseData {
+			return &AccessTokenSettings{}
+		},
+	}
+}
+
+func updateConfig() core.ResourceOperationConfig {
+	return core.ResourceOperationConfig{
+		Name:       "AccessTokenSettingsUpdate",
+		HttpMethod: http.MethodPut,
+		CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+			return fmt.Sprintf("https://%s/v1/accessTokens/settings", c.ControlPlane)
+		},
+		NewResourceData: func() core.ResourceData {
+			return &AccessTokenSettings{}
+		},
+	}
+}
+
+func resourceAccessTokenSettingsDelete(
+	ctx context.Context,
+	d *schema.ResourceData,
+	m interface{},
+) diag.Diagnostics {
+	// Since access token settings cannot be deleted, we just set the ID to
+	// empty so that the resource can be removed from the terraform state.
+	d.SetId("")
+	return diag.Diagnostics{}
+}
```

Comment:
@wcmjunior For the access token settings resource, since it is a global configuration that will always exist and never be actually deleted, I can think of two different approaches to handle the delete operation for this resource:

1. The delete operation would just set the resource ID to empty, so that the resource can be removed from the terraform state (this is how the resource is currently implemented in this PR). This will preserve the current token settings that are configured in CP.
2. The delete operation will set the token settings to the corresponding default values. The problem with this approach is that we will have to duplicate the default values here on terraform provider side, and if the default value change on the API and not on terraform provider, this would lead to unexpected behavior (because the terraform would use different default values to set the token settings on deletion).

I believe 1 would make more sense, since the settings are meant to just be read/updated, so operations like create/delete should only be a terraform concern in this case, such as creating or removing the resource from state.

Reply:
Let's discuss this internally before we move.

Reply:
As discussed, we opted for changing the delete operation to reset the access token settings to their corresponding default values.